src/components/MenuBar/MenuBar.tsx

```diff
@@ -50,17 +49,19 @@ export type MenuBarProps = {
  *
  * @category Components
  */
-const MenuBar = ({ useLearningPathTopic = _useLearningPathTopic }: MenuBarProps) => {
+
+const MenuBar = ({ courseSelected = false }: MenuBarProps) => {
   const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null)
   const [anchorElTopics, setAnchorElTopics] = useState<null | HTMLElement>(null)
   const { addSnackbar } = useContext(SnackbarContext)
   const { isAuth, logout } = useContext(AuthContext)
-  const userCourse = useStore((state) => state.course)
+  const { courseId } = useParams() as { courseId: string }
   const { t } = useTranslation()
-
-  //Application logic hooks
-  const { loading, topics } = useLearningPathTopic()
-  const reversedTopics: Topic[] = [...topics]
+  const [loadingTopics, setLoadingTopics] = useState(true)
+  const [topicsPath, setTopicsPath] = useState<Topic[]>([])
+  const fetchUser = usePersistedStore((state) => state.fetchUser)
+  const fetchLearningPathTopic = useStore((state) => state.fetchLearningPathTopic)
+  const reversedTopics: Topic[] = [...topicsPath]
   reversedTopics.sort((a, b) => reversedTopics.indexOf(b) - reversedTopics.indexOf(a))
 
```

Comment:
Objekte als immutable betrachen. Und dann wird instant ein einzeiler draus.
```suggestion
  const reversedTopics: Topic[] = [...topicsPath].sort((a, b) => reversedTopics.indexOf(b) - reversedTopics.indexOf(a))

```